Generated test bundle is broken on Patrol 2.0 on Windows

### Steps to reproduce

Create a simple Flutter project and setup patrol native automation.
main_test.dart: 

```dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:patrol/patrol.dart';

void main() {
  patrolTest(
    'counter state is the same after going to home and switching apps',
    nativeAutomation: true,
        ($) async {
      // Replace later with your app's main widget
      await $.pumpWidgetAndSettle(
        MaterialApp(
          home: Scaffold(
            appBar: AppBar(title: const Text('app')),
            backgroundColor: Colors.blue,
          ),
        ),
      );

      expect($('app'), findsOneWidget);
      await $.native.pressHome();
    },
  );
}
```

MainActivityTest.java:

```java
package com.ecample.test; // replace "pl.leancode.patrol.example" with your app's package

import androidx.test.platform.app.InstrumentationRegistry;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.junit.runners.Parameterized.Parameters;
import pl.leancode.patrol.PatrolJUnitRunner;

@RunWith(Parameterized.class)
public class MainActivityTest {
    @Parameters(name = "{0}")
    public static Object[] testCases() {
        PatrolJUnitRunner instrumentation = (PatrolJUnitRunner) InstrumentationRegistry.getInstrumentation();
        instrumentation.setUp(MainActivity.class);
        instrumentation.waitForPatrolAppService();
        return instrumentation.listDartTests();
    }

    public MainActivityTest(String dartTestName) {
        this.dartTestName = dartTestName;
    }

    private final String dartTestName;

    @Test
    public void runDartTest() {
        PatrolJUnitRunner instrumentation = (PatrolJUnitRunner) InstrumentationRegistry.getInstrumentation();
        instrumentation.runDartTest(dartTestName);
    }
}
```

build.gradle: 

```gradle
def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterRoot = localProperties.getProperty('flutter.sdk')
if (flutterRoot == null) {
    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"

android {
    namespace "com.example.test.int_test"
    compileSdkVersion flutter.compileSdkVersion
    ndkVersion flutter.ndkVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "com.example.test.int_test"
        // You can update the following values to match your application needs.
        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-gradle-build-configuration.
        minSdkVersion flutter.minSdkVersion
        targetSdkVersion flutter.targetSdkVersion
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
        testInstrumentationRunner "pl.leancode.patrol.PatrolJUnitRunner"
    }

    testOptions {
        execution "ANDROIDX_TEST_ORCHESTRATOR"
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }
}

dependencies {
    androidTestUtil "androidx.test:orchestrator:1.4.2"
}

flutter {
    source '../..'
}

```

### Actual results

This is the error i get
C:\Users\User\AndroidStudioProjects\int_test>patrol test --target integration_test/main_test.dart    
You're using Patrol CLI 2.0, which has breaking changes.
Read the migration guide at https://patrol.leancode.co/v2.
Disable this warning by setting the PATROL_MIGRATED environment variable.

No device specified, using the first one (emulator-5554)
• Building apk with entrypoint test_bundle.dart...
        integration_test/test_bundle.dart:12:8: Error: Error when reading '/main_test.dart': The system cannot find the file specified.

        import '/main_test.dart' as __main_test;
               ^
        integration_test/test_bundle.dart:65:35: Error: Undefined name 'main'.
          group('.main_test', __main_test.main);
                                          ^^^^
        Target kernel_snapshot failed: Exception


        FAILURE: Build failed with an exception.

        * Where:
        Script 'C:\Program Files\Flutter\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1201

        * What went wrong:
        Execution failed for task ':app:compileFlutterBuildDebug'.
        > Process 'command 'C:\Program Files\Flutter\flutter\bin\flutter.bat'' finished with non-zero exit value 1

        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.

        * Get more help at https://help.gradle.org

        BUILD FAILED in 37s
✗ Failed to build apk with entrypoint test_bundle.dart (Gradle build failed with code 1) (38.4s)
Exception: Gradle build failed with code 1
See the logs above to learn what happened. Also consider running with --verbose. If the logs still aren't useful, then it's a bug - please report it.
 FAIL  test_bundle.dart on emulator-5554

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>


### Patrol version

patrol: 2.0.0

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
C:\Users\User\AndroidStudioProjects\int_test>patrol doctor
You're using Patrol CLI 2.0, which has breaking changes.
Read the migration guide at https://patrol.leancode.co/v2.
Disable this warning by setting the PATROL_MIGRATED environment variable.

Patrol CLI version: 2.0.0
Program adb found in C:\Users\User\AppData\Local\Android\Sdk\platform-tools\adb.exe
Env var $ANDROID_HOME is not set
```

</details>


### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.5, on Microsoft Windows [Version 10.0.19044.1288], locale en-GB)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[√] Chrome - develop for the web
[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 2020.3)
[√] Android Studio (version 2022.2)
[√] Connected device (4 available)
[√] Network resources

! Doctor found issues in 1 category.


```

</details>

@bartekpacia maybe this helps. I tried changing a little bit the command and the issue dissapeared but another one came to life:
C:\Users\User\Documents\DashFuel\flutter>patrol test --target integration_test\main_test.dart 
You're using Patrol CLI 2.0, which has breaking changes.
Read the migration guide at https://patrol.leancode.co/v2.
Disable this warning by setting the PATROL_MIGRATED environment variable.

No device specified, using the first one (emulator-5554)
• Building apk with entrypoint test_bundle.dart...
        Invalid depfile: C:\Users\User\Documents\DashFuel\flutter\.dart_tool\flutter_build\15be9d42c5bda99275f2fd12e979251a\kernel_snapshot.d
        Invalid depfile: C:\Users\User\Documents\DashFuel\flutter\.dart_tool\flutter_build\15be9d42c5bda99275f2fd12e979251a\kernel_snapshot.d
        Unhandled exception:
        FileSystemException(uri=c:/Users/User/Documents/DashFuel/flutter/integration_test/main_test.dart; message=StandardFileSystem only supports file:* and data:* URIs)        
        #0      StandardFileSystem.entityForUri (package:front_end/src/api_prototype/standard_file_system.dart:34:7)
        #1      IncrementalCompiler._initializeExperimentalInvalidation (package:front_end/src/fasta/incremental_compiler.dart:1268:56)
        #2      IncrementalCompiler.computeDelta.<anonymous closure> (package:front_end/src/fasta/incremental_compiler.dart:325:17)
        <asynchronous suspension>
        #3      IncrementalCompiler.compile (package:vm/incremental_compiler.dart:75:50)
        <asynchronous suspension>
        #4      FrontendCompiler.compile (package:frontend_server/frontend_server.dart:595:11)
        <asynchronous suspension>
        #5      starter (package:frontend_server/starter.dart:99:12)
        <asynchronous suspension>
        #6      main (file:///C:/b/s/w/ir/x/w/sdk/pkg/frontend_server/bin/frontend_server_starter.dart:13:14)
        <asynchronous suspension>

        Target kernel_snapshot failed: Exception


        FAILURE: Build failed with an exception.

        * Where:
        Script 'C:\Program Files\Flutter\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1201

        * What went wrong:
        Execution failed for task ':app:compileFlutterBuildDebug'.
        > Process 'command 'C:\Program Files\Flutter\flutter\bin\flutter.bat'' finished with non-zero exit value 1

        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.

        * Get more help at https://help.gradle.org

        BUILD FAILED in 31s
✗ Failed to build apk with entrypoint test_bundle.dart (Gradle build failed with code 1) (32.7s)
Exception: Gradle build failed with code 1
See the logs above to learn what happened. Also consider running with --verbose. If the logs still aren't useful, then it's a bug - please report it.
 FAIL  test_bundle.dart on emulator-5554

--- a/packages/patrol_cli/CHANGELOG.md
+++ b/packages/patrol_cli/CHANGELOG.md
@@ -1,3 +1,7 @@
+## Unreleased
+
+- Fix bug with `test_bundle.dart` being sometimes garbled on Windows (#1797)
+
 ## 2.2.0
 
 - Add support for `group()` (#1634)

--- a/packages/patrol_cli/lib/src/test_bundler.dart
+++ b/packages/patrol_cli/lib/src/test_bundler.dart
@@ -177,7 +177,8 @@ ${generateGroupsCode([testFilePath]).split('\n').map((e) => '  $e').join('\n')}
     for (final testFilePath in testFilePaths) {
       final relativeTestFilePath = _normalizeTestPath(testFilePath);
       final testName = _createTestName(relativeTestFilePath);
-      imports.add("import '$relativeTestFilePath' as $testName;");
+      final relativeTestFilePathWithoutSlash = relativeTestFilePath[0] == '/' ? relativeTestFilePath.replaceFirst('/', '') : relativeTestFilePath;
+      imports.add("import '$relativeTestFilePathWithoutSlash' as $testName;");
     }
 
     return imports.join('\n');

--- a/packages/patrol_cli/lib/src/test_bundler.dart
+++ b/packages/patrol_cli/lib/src/test_bundler.dart
@@ -177,7 +177,9 @@ ${generateGroupsCode([testFilePath]).split('\n').map((e) => '  $e').join('\n')}
     for (final testFilePath in testFilePaths) {
       final relativeTestFilePath = _normalizeTestPath(testFilePath);
       final testName = _createTestName(relativeTestFilePath);
-      final relativeTestFilePathWithoutSlash = relativeTestFilePath[0] == '/' ? relativeTestFilePath.replaceFirst('/', '') : relativeTestFilePath;
+      final relativeTestFilePathWithoutSlash = relativeTestFilePath[0] == '/'
+          ? relativeTestFilePath.replaceFirst('/', '')
+          : relativeTestFilePath;
       imports.add("import '$relativeTestFilePathWithoutSlash' as $testName;");
     }
 

